### PR TITLE
DCAP driver: removed PLE and added new provision interface 

### PR DESCRIPTION
**Pull request by haimc-intel**

### CODE REVIEW

File: driver/linux/sgx_main.c [hunk 1]
```diff
@@ -309,14 +248,16 @@ static int sgx_pm_suspend(struct device *dev)
...
 static struct bus_type sgx_bus_type = {
 	.name	= "sgx",
 };
 
 struct sgx_context {
 	struct device dev;
 	struct cdev cdev;
+	struct device provision_dev;
+	struct cdev provision_cdev;
+	struct kobject *kobj_dir;
```
Comment (haitaohuang) [inline driver/linux/sgx_main.c]: Can we just attach the driver info to the driver/device node that's already in sysfs, e.g /bus/sgx or /module/intel_sgx?
The top dir of sys seems only contain top categories of things like bus, block, class, etc.

File: driver/linux/sgx_main.c [hunk 2]
```diff
@@ -371,54 +327,31 @@ static struct sgx_context *sgxm_ctx_alloc(struct device *parent)
...
+}
 
+static ssize_t sgx_store(struct kobject *kobj,
+					struct kobj_attribute *attr,const char *buf, size_t count)
+{
 	return 0;
 }
 
+struct kobj_attribute info_attr = __ATTR(info, 0444, sgx_info_show, sgx_store);
+struct kobj_attribute version_attr = __ATTR(version, 0444, sgx_version_show, sgx_store);
```
[the remark this answers is not in the document]
Comment (haimc-intel) [reply driver/linux/sgx_main.c]: Right, will fix.

File: driver/linux/sgx_main.c (hunk 1 above)
Comment (haimc-intel) [reply driver/linux/sgx_main.c]: The proposed path is /sys/kernel/sgx/, I think /module will be different between the OOT and DCAP driver.

**Comment by haimc-intel** [comment]
> Maybe we separate driver info parts into its own PR?

Each change is in different commit

**Pull request closed without merging**